pons/routing/router.py

```diff
@@ -68,7 +66,8 @@ def start(self, netsim: pons.NetSim, my_id: int):
         self.env.process(self.scan())
 
     def scan(self):
-        while True:
+        node = self.netsim.nodes[self.my_id]
+        while node.energy_model.energy > 0:
```

Comment:
why have this check here? isn't it enough to have an energy check in the send and receive functions of node.py so the router or any other upper layers are not relevant any more? ofc the scan loop can also contain the check to avoid some CPU cycles but deactivating the node functions is the safe bet to ensure that no further communication happens